qa/L0_backend_python/restart/models/restart/1/model.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import triton_python_backend_utils as pb_utils
+import c_python_backend_utils as c_utils
+from os import path
+
+
+class TritonPythonModel:
+
+    def execute(self, requests):
+        # This function will be called once to record the free memory. Then,
+        # the stub process will be killed to trigger Python backend restart.
+        # After that this value will be read again to make sure that it matches
+        # before restart.
+
+        file_name = 'free_memory.txt'
+        current_free_memory = str(c_utils.shared_memory.free_memory())
+        if path.exists(file_name):
+            with open(file_name, 'r') as f:
+                expected_free_memory = f.read()
+                assert(expected_free_memory == current_free_memory)
```

Comment:
Could you add a message to the assert to provide the values of expected_free_memory and current_free_memory?

Reply:
Improved the assert error message.

Reply:
Awesome! Approving, though you might want to change the verbiage (nit).  "is equal to X is not equal to" can be a bit confusing.